pkg/tlsutil/tls.go

```diff
@@ -158,14 +162,49 @@ func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service
 		return nil, nil, nil, err
 	}
 
+	var customCAKey *rsa.PrivateKey
```

Comment:
you can have 
```go
var (
caSecret *v1.Secret
caConfigMap *v1.ConfigMap
)
```
 here instead of `var customCAKey *rsa.PrivateKey`

And In your `if config.CAKey != "" && config.CACert != "" {` block, you can wrap `customCAKey` and `customCAKey` into `caSecret` and `caConfigMap` by calling `toCASecretAndConfigmap`. then the rest of code should take care of itself.

The steps can be:
1. check if custom ca is passed in.
2. if passed in, transform those into `caSecret` and `caConfigMap` in your `if config.CAKey != "" && config.CACert != ""` block.
3. if not passed, call `getCASecretAndConfigMapInCluster` to get CA from cluster.
4. rest of code doesn't need to be modify.

Reply:
but why would we want to transform a custom CA provided to us in PEM format to the type *v1.Secret and *v1.ConfigMap. Its a just an extra overhead